### ISSUE TITLE
Feature Request - Unity Cloud Build integration

### ISSUE THREAD

**Issue by efarraro** [google/play-unity-plugins#7]
This request requires at least some work on the Unity Cloud Build team and I'll reach out to them as well, but wanted to check in with this team as well.

Unity Cloud Build is a CI service provided by Unity that can build Unity projects easily, including support for building .aabs.

I noticed that in order to have asset packs build into the .aab (confirmed by inspecting the final .aab), you have to use the Google > Build Android App Bundle tool. If you build an .aab 'normally' using Build Settings > Build with the 'Build App Bundle' option checked, *that* .aab doesn't have the asset packs. That implies there must be something different about the Google specific approach vs the normal Unity approach for building .aab.

I saw in the docs for Asset Delivery that there is an API for building the .aab (https://developer.android.com/guide/app-bundle/asset-delivery/build-unity), although Unity Cloud Build just builds the .aab the 'normal' way. If there were a way to configure Unity to include the asset packs as part of the normal .aab build process somehow, that would be ideal.

Not sure if this is possible or if this is even the right place to integrate, but figured I'd ask about the difference between the 'normal' vs 'Google' option for building .aab in Unity.

**Comment by thanwah** [comment]
Thanks for creating this issue. I'm taking a look at Cloud Build and will let you know what I find out.

**Comment by thanwah** [comment]
After an initial investigation I think that Cloud Build's [advanced option](https://docs.unity3d.com/Manual/UnityCloudBuildAdvancedOptions.html) "Post-Export Method Name" should make it possible to eventually enable cloud built AABs with Asset Delivery enabled. However, it also may require a fairly big change relative to our current build approach. (I'll do a bit more investigation to confirm.)

Since I can think of a couple potential approaches, out of curiosity, what version of Unity are you using for your project's Cloud Build?

**Comment by efarraro** [comment]
Yeah - I was thinking of the Post Export method feature of Unity Cloud Build but came to the same conclusion. It doesn't seem like it's _currently_ possible, as best as I can tell.

I'm using Unity 2018.4 (I believe the LTS version)

**Comment by chbecher** [comment]
Hey, would it not be possible to use Gradle directly as described [here](https://developer.android.com/guide/app-bundle/asset-delivery/build-native-java) and let the Editor generate all needed files?

**Comment by s6-willard** [comment]
Is there an update on this? I'm also looking into the Post Export method, and the path it gives is a complete AAB file. I'm thinking it would be possible to unzip that AAB, then redo the bundle building with the asset packs part.

**Comment by VirsabiMunk** [comment]
+1

**Comment by LeshM** [comment]
+1
